File: api/application/usecase/responsemodel/beautician.go
package responsemodel

import (
	"time"
)

// Beautician response構造体
type Beautician struct {
	LineID      *string   `json:"lineId"`
	InstagramID *string   `json:"instagramId"`
	Comment     *string   `json:"comment"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
}

// BeauticianCreate response構造体
type BeauticianCreate struct {
	*Beautician `json:"beautician"`
}

// BeauticianGet response構造体
type BeauticianGet struct {
	*Beautician `json:"beautician"`
}

// BeauticianFind response構造体
type BeauticianFind struct {
	Beauticians []*User `json:"users"`
}

// BeauticianMyPageGet response構造体
type BeauticianMyPageGet struct {
	*User       `json:"user"`
	*Beautician `json:"beautician"`
	Salons      []*Salon `json:"salons"`
}

File: web/src/package/interface/response/Beautician.ts
import { BeauticianInfo } from "../Beautician";
import { BeauticianMenu } from "../Menu";
import { Salon } from "../Salon";
import { User } from "../User";

export interface BeauticianMyPage {
  user: User
  beautician: BeauticianInfo
  salons: Salon[]
  beauticianMenus: BeauticianMenu[]
}